### PR TITLE
docs(platform-browser): clarify TransferState usage

### PR DESCRIPTION
**Pull request by jorroll**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Documentation content changes
```

## What is the current behavior?

At the moment, using `TransferState` on the client generally requires changing the typical Angular app's bootstrap process to occur after `DOMContentLoaded`. See issue #20484.
If a client's bootstrap process occurs before `DOMContentLoaded`, the client fails to pick up the state passed from the server.

Unfortunately, there is no mention of this requirement in the API docs. This has generated some confusion for me and others (see https://stackoverflow.com/questions/47620181/angular-universal-transferstate-not-loading-on-client/47620452).

## What is the new behavior?

Updated the docs to reference issue #20484 and the fact that most project's will need to change their app's bootstrapping process. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by thefliik** [comment]
It would be great if this documentation update could be merged in. It will definitely help some users avoid a lot of frustration.

**Comment by brandonroberts** [comment]
@alan-agius4 is this change still needed?

**Comment by alan-agius4** [comment]
@brandonroberts, when using universal this is needed, and if one uses ng add this wrapping is done automaticlly.


See also: https://github.com/angular/universal/blob/master/docs/transfer-http.md

I am not actually sure of the value of adding this note though, as typically the users don't add `TransferState` themselves but rather it's added via `ServerTransferStateModule`.

**Comment by thefliik** [comment]
I still regularly get upvotes on a [stackoverflow answer (and question)](https://stackoverflow.com/a/47620452) I created on this topic. Is there any reason why it would be harmful to add this note to the API docs (so long as #20484 remains open)?

Using the angular cli is not a requirement for angular, so I don't think its a good idea to count on unspoken implementation details being handled by `ng add`.

I can definitely rebase this / make any changes that are needed. Perhaps additionally adding a note that this is not needed if someone imports `ServerTransferStateModule`? (which I did not know about, previously)

**Comment by brandonroberts** [comment]
Closing this one for now. Will revisit if it remains a recurring issue.

**Comment by thefliik** [comment]
@brandonroberts I'm still receiving +1 votes on [the S.O. answer](https://stackoverflow.com/a/47620452/5490505) I posted about this issue. Including this comment today:

> I added SSR support to my application using CLI, but it has this problem, it took me around 3 days to realize what was the problem. Thanks for this useful answer.

This sounds like 3 days of frustration that could have been avoided with better documentation.

I think another comment says it best:

> I wonder why Angular team doesn't include this information in their official document ([angular.io/api/platform-browser/TransferState](angular.io/api/platform-browser/TransferState))

It feels strange needing to justify, to this degree, a minor documentation improvement. Unlike @alan-agius4 suggests, clearly users do add `TransferState` directly. I also just realized that the docs @alan-agius4 linked to do not appear to be on angular.io. The decision to not improve the `angular.io` docs in this area feels unjustified. More practically, what's the argument in favor of *ever* not improving the docs? People learn differently, it's helpful to say the same thing multiple times in different places and in different ways.

**Comment by m98** [comment]
@brandonroberts I totally agree with this PR as it's going to improve the documentation. This problem itself took my time, it should be fixed, both in CLI (when creating a new SSR) and also in documentation.

File: packages/platform-browser/src/browser/transfer_state.ts [hunk 1]
```diff
@@ -76,6 +76,11 @@ export function makeStateKey<T = void>(key: string): StateKey<T> {
  * boolean, number, string, null and non-class objects will be serialized and deserialzied in a
  * non-lossy manner.
  *
+ * *Note:*
+ * *While using `TransferState` on the client, it may be necessary to update the client's*
+ * *bootstrap process (generally in `main.ts`) to wait for `DOMContentLoaded`. See angular repo*
+ * *issue [#20484](https://github.com/angular/angular/issues/20484) for more info.*
```
Comment (m98) [inline packages/platform-browser/src/browser/transfer_state.ts]: I think this type of language is not what we have in other parts of Angular documentation. We need to clarify in a better way. Just show the code that they need to add to the `main.ts`

**Pull request closed without merging**